fatiando/gridder.py

```diff
@@ -439,3 +444,269 @@ def cut(x, y, scalars, area):
+    npt = []
+    nd = a.ndim
+    if npd == None:
+        for ii in range(0, nd):
+            if nd == 1:
+                npt.append(_nextpow2(len(a)))
+            else:
+                npt.append(_nextpow2(a.shape[ii]))
+    else:
+        et = 'gridder.pad_array: pad dimensions do not match array dims'
+        if nd == 1:
+            if _is_integer(npd):
+                npt.append(npd)
+            else:
+                if len(npd) != 1:
+                    raise ValueError(et)
+                else:
+                    npt.append(npd)
+        else:
+            if _is_integer(npd):
+                raise ValueError(et)
+            elif len(npd) != a.ndim:
+                raise ValueError(et)
+            else:
+                npt = npd
+        for ii in numpy.arange(len(npt)):
+            if npt[ii] <= a.shape[ii]:
+                # How are you supposed to properly assign a multi-line string?
+                et1 = 'gridder.pad_array: Desired padding is less than array '
+                et2 = 'length along dimension '+str(ii)
+                et = et1+et2
+                raise ValueError(et)
+    # Compute numbers to pad on the left and right side of the array along
+    # each dimension
+    nps = []
+    for ii in range(0, nd):
+        nps.append((int(numpy.ceil((npt[ii] - a.shape[ii])/2.)),
+            int(numpy.floor((npt[ii] - a.shape[ii])/2.))))
+
+    # If it will be needed, compute the mean
+    meanneeded = ['lintaper', 'oddreflectiontaper']
+    if padtype.lower() in meanneeded:
+        m = numpy.mean(a)
+    # Use numpy's padding routines where possible
+    if _is_number(padtype):
+        # Pad with value
+        ap = numpy.pad(numpy.copy(a), nps, mode='constant',
+            constant_values = float(padtype))
+    elif padtype.lower() == 'mean':
+        # Pad with the mean
+        ap = numpy.pad(numpy.copy(a), nps, mode='mean')
+    elif padtype.lower() == 'lintaper':
+        # Linearly taper to the mean
+        ap = numpy.pad(numpy.copy(a), nps, mode='linear_ramp', end_values=m)
+    elif padtype.lower() == 'edge':
+        # Pad with edge values
+        ap = numpy.pad(numpy.copy(a), nps, mode='edge')
+    elif padtype.lower() == 'reflection':
+        # Pad with even reflection 
+        ap = numpy.pad(numpy.copy(a), nps, mode='reflect', 
+            reflect_type='even')
+    elif padtype.lower() == 'oddreflection':
+        # Pad with odd reflection
+        ap = numpy.pad(numpy.copy(a), nps, mode='reflect', 
+            reflect_type='odd')
+    elif padtype.lower() == 'oddreflectiontaper':
+        # Pad with odd reflection and a cosine taper to mean
+        ap = (numpy.pad(numpy.copy(a), nps, mode='reflect', 
+            reflect_type='odd') - m)
+        for ii in numpy.arange(nd):
+            ap = numpy.apply_along_axis(_costaper, ii, ap, lp=nps[ii][0],
+                rp=nps[ii][1])
+        ap += m
+    if xy is not None:
+        cp = _padcoords(xy, a.shape, nps)
```

Comment:
Hum. This makes me think that maybe it would be simpler to just make `_padcoords` into a public `pad_coords` function that does the extending of the coordinate arrays. That would greatly simplify the code, calling and returning of this function.